functions/functionsv2/label_gce_instance/label_gce_instance.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_label_gce_instance]
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/googleapis/gax-go/v2"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// AuditLogEntry represents a LogEntry as described at
+// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
+type AuditLogEntry struct {
+	ProtoPayload *AuditLogProtoPayload `json:"protoPayload"`
+}
+
+// AuditLogProtoPayload represents AuditLog within the LogEntry.protoPayload
+// See https://cloud.google.com/logging/docs/reference/audit/auditlog/rest/Shared.Types/AuditLog
+type AuditLogProtoPayload struct {
+	MethodName         string                 `json:"methodName"`
+	ResourceName       string                 `json:"resourceName"`
+	AuthenticationInfo map[string]interface{} `json:"authenticationInfo"`
+}
+
+//Interface for the client, so we can mock it out in tests.
+type GceInstancesClient interface {
+	Get(context.Context, *computepb.GetInstanceRequest, ...gax.CallOption) (*computepb.Instance, error)
+	SetLabels(context.Context, *computepb.SetLabelsInstanceRequest, ...gax.CallOption) (*compute.Operation, error)
+}
+
+var client GceInstancesClient
+
+func init() {
+	// Create an Instances Client
+	var err error
+	client, err = compute.NewInstancesRESTClient(context.Background())
+	if err != nil {
+		log.Fatalf("Failed to create instances client: %s", err)
+	}
+}
+
+// Cloud Function that receives GCE instance creation Audit Logs, and adds a
+// `creator` label to the instance.
+func LabelGceInstance(ctx context.Context, e event.Event) error {
+	// Extract parameters from the Cloud Event and Cloud Audit Log data
+	logentry := &AuditLogEntry{}
+	if err := e.DataAs(logentry); err != nil {
+		werr := fmt.Errorf("event.DataAs() : %w", err)
+		log.Printf("Error parsing proto payload: %s", werr)
+		return werr
+	}
+	payload := logentry.ProtoPayload
+	creator, ok := payload.AuthenticationInfo["principalEmail"]
+	if !ok {
+		e := fmt.Errorf("principalEmail not found in cloud event payload: %v", payload)
```

Comment:
suggestion: Given how common it is to use `e` as a secondary name for `err`, maybe we should change the variable representing event to `ce` or `ev`?

Reply:
good call. fixed.